src/launch/cli/service/publish/__init__.py

```diff
@@ -0,0 +1,235 @@
+                    fg="red",
+                )
+                quit()
+            else:
+                temp_server_url = readFile("GIT_SERVER_URL")
+                temp_org = readFile("GIT_ORG")
+                temp_repo = readFile("GIT_REPO")
+                url = f"{temp_server_url}/{temp_org}/{temp_repo}"
+                tag = readFile("MERGE_COMMIT_ID")
+                os.environ["CONTAINER_IMAGE_VERSION"] = readFile(
+                    "CONTAINER_IMAGE_VERSION"
+                )
+                service_dir = service_dir.joinpath(extract_repo_name_from_url(url))
+        else:
+            with open(LAUNCHCONFIG_NAME, "r") as f:
+                input_data = json.load(f)
+                url = input_data["sources"]["application"]["url"]
+                tag = input_data["sources"]["application"]["tag"]
+                service_dir = service_dir.joinpath(extract_repo_name_from_url(url))
+
+    if not skip_clone:
+        repository = clone_repository(
+            repository_url=url,
+            target=service_dir,
+            branch=SERVICE_MAIN_BRANCH,
+            dry_run=dry_run,
+        )
+        checkout_branch(
+            repository=repository,
+            target_branch=tag,
+            dry_run=dry_run,
+        )
+
+    if GITHUB_PUBLISH_TOKEN_SECRET_NAME is None:
+        click.secho(
+            f"No PAT token found for AWS CodeBuild. Please set {GITHUB_PUBLISH_TOKEN_SECRET_NAME} in the environment variables.",
+            fg="red",
+        )
+        quit()
+    else:
+        package_publisher, package_registry, package_scope, repo_path, source_branch = set_github_variables(
+            package_publisher, package_registry, package_scope, repo_path, source_branch
+        )
+
+        execute_publish(
+            service_dir=service_dir,
+            provider=provider,
+            dry_run=dry_run,
+            token_secret_name=GITHUB_PUBLISH_TOKEN_SECRET_NAME,
+            package_scope=package_scope,
+            package_publisher=package_publisher,
+            package_registry=package_registry,
+            source_folder_name=source_folder_name,
+            repo_path=repo_path,
+            source_branch=source_branch,)
+
+def set_github_variables(package_publisher, package_registry, package_scope, repo_path, source_branch):
+    if GITHUB_PACKAGE_PUBLISHER:
+        package_publisher = GITHUB_PACKAGE_PUBLISHER
+    if GITHUB_PACKAGE_REGISTRY:
+        package_registry = GITHUB_PACKAGE_REGISTRY
+    if GITHUB_PACKAGE_SCOPE:
+        package_scope = GITHUB_PACKAGE_SCOPE
+    if GITHUB_REPO_PATH:
+        repo_path = GITHUB_REPO_PATH
+    if GITHUB_SOURCE_BRANCH:
+        source_branch = GITHUB_SOURCE_BRANCH
+
+    return package_publisher, package_registry, package_scope, repo_path, source_branch
```

Comment:
Can you help me understand the choice behind this function as opposed to setting defaults on these values using the @click.option decorators?

Reply:
removed function!! Added defaults!!